ctest-next/templates/test.rs

```diff
@@ -82,7 +83,430 @@ mod generated_tests {
+    fn roundtrip_padding_{{ ident }}() -> Vec<u8> {
+        // stores (offset, size) for each field
+        {#- If there is no public field these become unused. +#}
+        {%- if fields.len() > 0 +%}
+        let mut v = Vec::<(usize, usize)>::new();
+        let bar = std::mem::MaybeUninit::<{{ ident }}>::uninit();
+        let bar = bar.as_ptr();
+        {%- else +%}
+        let v = Vec::<(usize, usize)>::new();
+        {%- endif +%}
+        {%- for field in fields +%}
+        unsafe {
+            let ty_ptr = std::ptr::addr_of!((*bar).{{ field.ident() }});
+            let val = ty_ptr.read_unaligned();
+            let size = mem::size_of_val(&val);
+            let off = offset_of!({{ ident }}, {{ field.ident() }});
+            v.push((off, size));
+        }
+        {%- endfor +%}
+        // This vector contains `1` if the byte is padding
+        // and `0` if the byte is not padding.
+        let mut pad = Vec::<u8>::new();
+        // Initialize all bytes as:
+        //  - padding if we have fields, this means that only
+        //  the fields will be checked
+        //  - no-padding if we have a type alias: if this
+        //  causes problems the type alias should be skipped
+        pad.resize(mem::size_of::<{{ ident }}>(), 1);
+        for (off, size) in &v {
+            for i in 0..*size {
+                pad[off + i] = 0;
+            }
+        }
+        pad
+    }
+
+    /// Tests whether the type alias `x` when passed to C and back to Rust remains unchanged.
+    ///
+    /// It checks if the size is the same as well as if the padding bytes are all in the
+    /// correct place.
+    pub fn roundtrip_{{ ident }}() {
+        use std::ffi::c_int;
+
+        type U = {{ ident }};
+        extern "C" {
+            fn __test_roundtrip_{{ ident }}(
+                size: i32, x: U, e: *mut c_int, pad: *const u8
+            ) -> U;
+        }
+        let pad = roundtrip_padding_{{ ident }}();
+        unsafe {
+            use std::mem::{MaybeUninit, size_of};
+            let mut error: c_int = 0;
+            let mut y = MaybeUninit::<U>::uninit();
+            let mut x = MaybeUninit::<U>::uninit();
+            let x_ptr = x.as_mut_ptr().cast::<u8>();
+            let y_ptr = y.as_mut_ptr().cast::<u8>();
+            let sz = size_of::<U>();
+            for i in 0..sz {
+                let c: u8 = (i % 256) as u8;
+                let c = if c == 0 { 42 } else { c };
+                let d: u8 = 255_u8 - (i % 256) as u8;
+                let d = if d == 0 { 42 } else { d };
+                x_ptr.add(i).write_volatile(c);
+                y_ptr.add(i).write_volatile(d);
+            }
+            let r: U = __test_roundtrip_{{ ident }}(sz as i32, x.assume_init(), &mut error, pad.as_ptr());
+            if error == 1 {
+                FAILED.store(true, Ordering::Relaxed);
+                return;
+            }
+            for (i, elem) in pad.iter().enumerate().take(size_of::<U>()) {
```

Comment:
Isn't `pad` already `size_of::<U>()`? Maybe this is something worth asserting, then you can remove the `.take()`